pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -256,7 +373,33 @@ func (service *SubnetPortService) DeleteSubnetPort(nsxSubnetPort *model.VpcSubne
 	return nil
 }
 
+// DeleteDHCPStaticBinding deletes DHCP static binding.
+func (service *SubnetPortService) DeleteDHCPStaticBinding(subnetPortInfo servicecommon.VPCResourceInfo, bindingId string) error {
+	staticBinding := service.DHCPStaticBindingStore.GetByKey(bindingId)
+	if staticBinding != nil {
+		err := service.NSXClient.DhcpStaticBindingConfigsClient.Delete(subnetPortInfo.OrgID, subnetPortInfo.ProjectID, subnetPortInfo.VPCID, subnetPortInfo.ParentID, bindingId)
+		err = nsxutil.TransNSXApiError(err)
+		if err != nil {
+			log.Error(err, "Failed to delete DHCP static binding", "dhcpStaticBinding.Id", bindingId)
+			return err
+		}
+		service.DHCPStaticBindingStore.Delete(staticBinding)
+	}
+	return nil
+}
+
 func (service *SubnetPortService) DeleteSubnetPortById(portID string) error {
+	dhcpStaticBinding := service.DHCPStaticBindingStore.GetByKey(portID)
```

Comment:
As DeleteSubnetPortById also calls DeleteSubnetPort, maybe we can just rely on the code in DeleteSubnetPort to delete the dhcpStaticBinding?

Reply:
Good point. Use DeleteSubnetPort to delete the dhcpStaticBinding.